p2p/server/server.go

```diff
@@ -240,12 +246,18 @@ func (s *Server) Run(ctx context.Context) error {
 				return nil
 			}
 			eg.Go(func() error {
+				conn := req.stream.Conn()
 				if s.decayingTag != nil {
-					s.decayingTag.Bump(req.stream.Conn().RemotePeer(), s.decayingTagSpec.Inc)
+					s.decayingTag.Bump(conn.RemotePeer(), s.decayingTagSpec.Inc)
 				}
 				ok := s.queueHandler(ctx, req.stream)
+				took := time.Since(req.received)
```

Comment:
Same here, `took` is an unusual variable name:

```suggestion
				duration := time.Since(req.received)
```

Reply:
Again, both `Duration` and `float64` (seconds) values are used in the code below

Reply:
Yes. They could be `duration` and `duration.Seconds()` instead of `took` and `took.Seconds()`?

Reply:
Done